tests/integration/kubernetes/k8s-policy-job.bats

```diff
@@ -9,7 +9,7 @@ load "${BATS_TEST_DIRNAME}/../../common.bash"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
 
 setup() {
-    policy_tests_enabled || skip "Policy tests are disabled."
+    auto_generate_policy_enabled || skip "Auto-generated policy tests are disabled."
```

Comment:
Do these changes mean that the tests with only run against mariner now due to:
https://github.com/kata-containers/kata-containers/blob/89c85dfe846d661dba0f610260a8f69c61aca967/tests/integration/kubernetes/gha-run.sh#L261-L269

whereas before they were running on snp, sev & tdx as well due to
https://github.com/kata-containers/kata-containers/blob/89c85dfe846d661dba0f610260a8f69c61aca967/tests/integration/kubernetes/tests_common.sh#L266-L271
?

Obviously I'd like to see them running on all platforms, but that can come as part of the future discussion about enabling policy for the vanilla agent :)

Reply:
@stevenhorsman 

- Mariner, SEV, SNP and TDX Hosts have been testing using just simple, hard-coded policies from https://github.com/kata-containers/kata-containers/tree/main/src/kata-opa since September (see https://github.com/kata-containers/kata-containers/commit/bf21411e90327b4c3192f1c1debe9cb0b1c4dcb5).

- Mariner started testing a more sophisticated, auto-generated, policy since February (https://github.com/kata-containers/kata-containers/commit/0de407f8b73ed2caf69205b6b859fbfa45a084f2). This testing took place just on Mariner because:
  - It's easier for me to connect and debug on Mariner Hosts when the tests fail (see below for the diagnosability improvements needed to address this).
  - Even while testing a single type of Host, developing tests and the genpolicy tool required by these tests, and passing CI has been challenging. *The policy tests are finally in pretty good shape just since last week*.

- Starting 2 weeks ago, I *accidentally* enabled new genpolicy tests for SEV, SNP and TDX Hosts too (https://github.com/kata-containers/kata-containers/commit/2252490a9688814c2f73b7bba1beb8d59973b33e)
  - SEV, SNP and TDX Hosts were failing CI for unrelated reasons, so none of us noticed this accident until Fabiano noticed yesterday.
  - Here, I am just fixing this accident: https://github.com/kata-containers/kata-containers/pull/9510/commits/e5c3f5fa9b4017b4892c3855701347ebbb8da9f5
  - So, these tests were enabled for a couple of weeks on SEV, SNP and TDX Hosts but never passed on these unexpected Hosts.

- Support for testing SEV, SNP and TDX and any other Hosts that might be interested in policy is coming up very soon. There were two requirements before enabling these tests:
  - Tests that are stable and have pretty good coverage. We satisfied this requirement just last week (https://github.com/kata-containers/kata-containers/commit/2c4d1ef76bf640d644dba26c97642afe49ddec05).
  - Policy logs that we can use to investigate failures, without me having to connect and debug on the Hosts. We'll finally get those logs from the current PR (https://github.com/kata-containers/kata-containers/pull/9510)

So, proper testing for SEV, SNP and TDX Hosts is coming up soon, but it depends on the current PR (https://github.com/kata-containers/kata-containers/pull/9510)

Reply:
Sounds good to me :)